app/models/calculated_handover_date.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+class CalculatedHandoverDate < ApplicationRecord
+  belongs_to :case_information,
+             primary_key: :nomis_offender_id,
+             foreign_key: :nomis_offender_id,
+             inverse_of: :responsibility
+
+  validates :nomis_offender_id, uniqueness: true, presence: true
+  validates :reason, presence: true
+
+  after_save :push_to_delius
+
+  def self.recalculate_for(offender)
+    record = self.find_or_initialize_by(nomis_offender_id: offender.offender_no)
```

Comment:
think find_or_initialize_by takes a block, so this could be a bit more idiomatic

Reply:
I did try this at first, but not sure it works for our needs here.

`find_or_initialize_by` only runs the in the case of 'initialize', whereas we want to update the attributes regardless of whether the record was 'found' or 'initialized'.

Reply:
@starswan To make this method more idiomatic, I've just replaced the use of `assign_attributes` + `save!` with `update!` instead.